scripts/docker_build/sof_builder/Dockerfile

```diff
@@ -111,21 +111,30 @@ ENV PATH="/home/sof/work/xtensa-mt8186-elf/bin:${PATH}"
 ENV PATH="/home/sof/work/xtensa-mt8195-elf/bin:${PATH}"
 
 ARG NEWLIB_REPO=https://github.com/jcmvbkbc/newlib-xtensa.git
-RUN cd /home/sof && \
+RUN cd "$HOME" && \
 	git clone $CLONE_DEFAULTS --branch xtensa $NEWLIB_REPO && \
 	cd newlib-xtensa && \
 	for arch in byt hsw apl cnl imx imx8m imx8ulp rn mt8186 mt8195; do \
 		./configure --target=xtensa-${arch}-elf \
 		--prefix=/home/sof/work/xtensa-root && \
-		make && \
-		make install && \
+		make && make install && \
 		rm -rf etc/config.cache; \
 	done && \
-	cd /home/sof/ && rm -rf newlib-xtensa.git
+	echo "Stage3: newlib-xtensa is done!"
 
+# log all hashes for each repo to a file
+ARG SOF_WORK="$HOME"/work
+ARG REPOS="$SOF_WORK/alsa/alsa-lib $SOF_WORK/alsa/alsa-utils $HOME/xtensa-overlay $HOME/crosstool-ng $HOME/newlib-xtensa"
+ARG HASH_LIST="$SOF_WORK"/sof_docker_hash.txt
+RUN echo "Build date: $(date +%Y%m%d)" > "$HASH_LIST" && set -e && \
+        for repo in $REPOS; do \
+                cd "$repo";  pwd >> "$HASH_LIST"; git log --oneline -5 >> "$HASH_LIST"; \
```

Comment:
```suggestion
                cd "$repo";  pwd; git log --oneline -5; done >> "$HASH_LIST"; \
```

Reply:
why 'done' is middle of the command?
and I'd like to keep the path also in the log

Reply:
This redirects the entire `for... done` loop once.

Reply:
Ah, I see, multiple commands in one line fool me. Good suggestion! Will upload it shortly

Reply:
Updated. Sorry for the delay